src/topology/editor.rs

```diff
@@ -308,8 +346,19 @@ impl<'topology> TopologyEditor<'topology> {
     ///
     /// # Errors
     ///
-    /// - [`AllowSetError`] if an `AllowSet::Custom` with both `cpuset` and
-    ///   `nodeset` set to `None` is passed in.
+    /// - [`EmptyCustom`] if an `AllowSet::Custom` does not do anything because
+    ///   both its `cpuset` and `nodeset` members are empty.
+    /// - [`InvalidCpuset`] if applying the `cpuset` of an `AllowSet::Custom`
+    ///   would amount to disallowing all CPUs from the topology.
+    /// - [`InvalidNodeset`] if applying the `nodeset` of an `AllowSet::Custom`
+    ///   would amount to disallowing all NUMA nodes from the topology.
+    /// - [`Unsupported`] if the specified `AllowSet` is not supported by the
+    ///   host operating system.
+    ///
+    /// [`EmptyCustom`]: AllowSetError::EmptyCustom
+    /// [`InvalidCpuset`]: AllowSetError::InvalidCpuset
+    /// [`InvalidNodeset`]: AllowSetError::InvalidNodeset
+    /// [`Unsupported`]: AllowSetError::Unsupported
```

Comment:
Test more error cases so that we depend less on the (low-quality) hwloc error reporting.